### ISSUE TITLE
plugin-search: Search results won't filter by Search_doc_Relation_RelationTo

### ISSUE THREAD

**Issue by slavanossar** [payloadcms/payload#5338]
### Describe the Bug

I am writing a query to retrieve search results that belong to a specific collection

```gql
query GetSearchResults(
  $searchTerm: String!
  $collectionSlug: Search_doc_Relation_RelationTo!
  $limit: Int
  $page: Int
) {
  Searches(
    where: {
      AND: [
        { title: { contains: $searchTerm } }
        { doc: { relationTo: $collectionSlug } }
      ]
    }
    limit: $limit
    page: $page
  ) {
    docs {
      id
      title
    }
  }
}
```

When executing the query for a specified `$collectionSlug`, the results include documents that are from all collections

<img width="1277" alt="image" src="https://github.com/payloadcms/payload/assets/7077600/8c7e3f0c-6867-40e5-aa18-4c4c71e4cbe2">

When a value that doesn't exist on `Search_doc_Relation_RelationTo` is used, the response correctly states that the value isn't part of the enum.

<img width="1277" alt="image" src="https://github.com/payloadcms/payload/assets/7077600/8260ca55-71f7-4574-9f92-71ff094e9bfd">

### To Reproduce

1. Create two collections
2. Enable search plugin for both collections
3. Create some docs
4. Perform a GraphQL query as described above

### Payload Version

2.11.2

### Adapters and Plugins

@payloadcms/plugin-search@1.1.0

**Comment by Sreesruth** [comment]
hii @slavanossar @denolfe @colinramsay @jmas @jschuur , i would like to contribute to this issue, thankyou

**Comment by wkentdag** [comment]
+1 experiencing the same issue. 

```
  "dependencies": {
    "@azure/storage-blob": "^12.17.0",
    "@nouance/payload-better-fields-plugin": "^1.3.5",
    "@payloadcms/bundler-webpack": "^1.0.6",
    "@payloadcms/db-postgres": "^0.x",
    "@payloadcms/plugin-cloud-storage": "^1.1.2",
    "@payloadcms/plugin-search": "^1.1.0",
    "@payloadcms/richtext-lexical": "^0.7.0",
    "cross-env": "^7.0.3",
    "dotenv": "^8.2.0",
    "express": "^4.17.1",
    "payload": "^2.11.2"
  },
```


<img width="1170" alt="Screen Shot 2024-04-05 at 5 58 23 PM" src="https://github.com/payloadcms/payload/assets/3254957/036a85f4-5693-49c5-9f6a-4cec8bf06c54">

**Comment by slavanossar** [comment]
I did some investigating and it seems like the issue actually originates within the `db-mongodb` adaptor, although haven't been able to pinpoint where.

**Comment by slavanossar** [comment]
Looking at the code in `db-mongodb` it's clear that `relationTo` isn't a supported operation and so is completely ignored by [`parseParams`](https://github.com/payloadcms/payload/blob/main/packages/db-mongodb/src/queries/parseParams.ts), [`buildSearchParam`](https://github.com/payloadcms/payload/blob/main/packages/db-mongodb/src/queries/buildSearchParams.ts), etc., due to not being a included in [`validOperators`](https://github.com/payloadcms/payload/blob/main/packages/payload/src/types/constants.ts) or in the [`operatorMap`](https://github.com/payloadcms/payload/blob/main/packages/db-mongodb/src/queries/operatorMap.ts).

It seems like the issue actually stems from the generation of the collection's schema, since `relationTo` should actually contain an operator within a `where` query.

The `Search_where` inputs are defined as

```graphql
input Search_where {
  title: Search_title_operator
  priority: Search_priority_operator
  doc: Search_doc_Relation
  updatedAt: Search_updatedAt_operator
  createdAt: Search_createdAt_operator
  id: Search_id_operator
  AND: [Search_where_and]
  OR: [Search_where_or]
}

input Search_doc_Relation {
  relationTo: Search_doc_Relation_RelationTo
  value: JSON
}

enum Search_doc_Relation_RelationTo {
  articles
  artists
  episodes
  hosts
  programs
  trackPlays
}
```

where in reality it should be something like

```graphql
input Search_where {
  title: Search_title_operator
  priority: Search_priority_operator
  doc: Search_doc_Relation
  updatedAt: Search_updatedAt_operator
  createdAt: Search_createdAt_operator
  id: Search_id_operator
  AND: [Search_where_and]
  OR: [Search_where_or]
}

input Search_doc_Relation {
  relationTo: Search_doc_Relation_RelationTo_operator
  value: JSON
}

input Search_doc_Relation_RelationTo_operator {
  equals: Search_doc_Relation_RelationTo
  not_equals: Search_doc_Relation_RelationTo
  like: Search_doc_Relation_RelationTo
  contains: Search_doc_Relation_RelationTo
  in: [Search_doc_Relation_RelationTo]
  not_in: [Search_doc_Relation_RelationTo]
  all: [Search_doc_Relation_RelationTo]
  exists: Boolean
}

enum Search_doc_Relation_RelationTo {
  articles
  artists
  episodes
  hosts
  programs
  trackPlays
}
```

**Comment by slavanossar** [comment]
Closed in favour of #6399
